### ISSUE TITLE
migrate to Urho3D 1.5

### ISSUE THREAD

**Issue by bvanevery** [gawag/Urho-Sample-Platformer#12]
Urho3D 1.5 has shipped!  It contains a lot of bug fixes and improvements to the CMake build system that make life easier, particularly on Windows.  Some of this is a byproduct of bugs I've been filing over the past 6 months, which the Urho3D developers have been fixing.  From the standpoint of someone trying to test USP, it would be highly desirable to have USP building against 1.5 rather than 1.4, as things like D3DCOMPILER get automatically detected according to correct Windows SDK, etc.  Saves a lot of configuration hassle.

**Comment by damu** [comment]
... and I just updated everything to 1.4...
Lets see if I can update everything to 1.5 in more reasonable time as the 1.4 updates.

**Comment by bvanevery** [comment]
#9 is the compile blocker for updating to 1.5.  The problem of WIN32 vs. _WIN32 could be causing that behavior.  https://github.com/urho3d/Urho3D/issues/1036  I haven't checked whether that's true though.  It would be best to iron that out first though, before doing much head scratching about 1.5.

**Comment by bvanevery** [comment]
I have noted how to fix the compile errors in #9, at least against Urho3D post-1.5 master branch.  The changes are fairly trivial.  I don't currently have a patch to offer as my Git skills are still poor.  But it shouldn't be hard to follow my notes on what's wrong and how to fix things.

Now there is some new handling of resource paths to contend with.  This takes some decision making about what ResourcePrefixPaths should be used in the build tree vs. the installation tree.  An example is provided in the Urho3D samples, in the file Urho3D\Source\Samples\Sample.inl in the Setup() function:

```
    // Construct a search path to find the resource prefix with two entries:
    // The first entry is an empty path which will be substituted with program/bin directory -- this entry is for binary when it is still in build tree
    // The second entry is a relative path from the installed program/bin directory to the asset directory -- this entry is for binary when it is in the Urho3D SDK installation location
    if (!engineParameters_.Contains("ResourcePrefixPaths"))
        engineParameters_["ResourcePrefixPaths"] = ";../share/Urho3D/Resources";
```

The approach taken when building Urho3D samples, is to copy the needed resources into the CMake build tree.  Then resources are copied again to the final installation tree.  This may seem a little piggy, but it's practical as the search is done with relative directories.  When doing a CMake "out of source" build, there is no absolute relationship between a source and a build tree, they could be anywhere.

An alternate approach would be to hardwire the absolute path of the source tree, using a CMake substitution.  I'm not sure that either is better or worse.  Both have to contend with the reality that it's a CMake build, with different directory relationships at different stages of the build.

**Comment by damu** [comment]
Update to Urho 1.5 finished.
